manager2/src/app/tps/tps.component.ts

```diff
@@ -225,15 +231,16 @@ export class TpsComponent implements OnInit {
     extend_reservation() {
         this.msg = '';
         this.errmsg = '';
-        if(new Date(this.new_expire).getTime() < this.selectedEvent.meta.end) {
+        const new_to_Date = new Date(this.new_toDate).getTime();
+        if(new_to_Date < this.selectedEvent.meta.end) {
             this.errmsg = 'Extended end date must be after current end date';
             return;
         }
-        if(new Date(this.new_expire).getTime() < new Date().getTime()) {
+        if(new_to_Date < new Date().getTime()) {
```

Comment:
Need a check for lock_date too (both cannot be in the past, and must be <= end date)